lib/stack/src/layer.rs

```diff
@@ -31,22 +32,6 @@ pub trait Layer<T, U, M: super::Stack<U>> {
             _p: PhantomData,
         }
     }
-
-    /// Conditionally compose this `Layer with another.
-    fn and_when<P, N, L>(self, predicate: P, inner: L)
```

Comment:
Since this method (and `Optional`) weren't being used, I think removing them was a good call. But it's good to know that they're possible if we end up needing them.